### ISSUE TITLE
Can't compile on macOS on M1 processor

### ISSUE THREAD

**Issue by profludic** [lronaldo/cpctelera#162]
Error:

*** BFD does not support target arm-apple-darwin21.6.0.
*** Look in bfd/config.bfd for supported targets.
make[4]: *** [configure-binutils] Error 1

Note:

I managed to compile it somehow by adding some of the changes from latest sdcc.

**Comment by lronaldo** [comment]
Hi @profludic 

Sorry for not answering earlier (didn't receive a notifcation).

Did you solve this issue? Can you share what you learnt while solving it? That would be very interesting :)

Thank you :)

**Comment by profludic** [comment]
Hello @lronaldo yes, I have solved it.

I forgot exactly what I did but in general it was solved by using latest sdcc and trying to make it compile - I think it is some kind of "Frankenstein" sdcc - mix between cpctelera version and newer version :) 

Since I just wanted to make it work, it is likely that I made some mistake but - it works for me...

I would love to have this in official cpctelera release properly done!

So, here is the source I use which is modified development branch of cpctelera (not sure how old is it): 

http://www.osmobit.com/sdcc.patched.zip

You can download it and compare to see all what is changed.

What I did after was:
cp -r sdcc.patched/* ./cpctelera/tools/sdcc-3.6.8-r9946

I hope this helps.

teopl

**Comment by lronaldo** [comment]
Thank you very much for contributing your sources.

The problem for me is that I don't have a MacOS available computer, so I cannot test and know what are the exact changes required for MacOS.

In any case, I'm not completely sure if it is more advisable to just update CPCtelera to the new SDCC version and test if that version properly compiles on MacOS and the newest M1 and M2 ARM CPUs.

Be it as it might, thank you very much. I'm sure that your source can help many other people interested in compiling under MacOS :)

I will try first with the update strategy and then we will see what to do next. If you like, you may test next updated version once we publish it and suggest changes for MacOS :)

Thank you again for your contribution.

**Comment by profludic** [comment]
Hello @lronaldo, I am glad I can help even a little since I love using cpctelera!

It could be that strategy of updating sdcc is the right way. For some reason I remember I did only partial update.

I would be happy to test new version!

teopl

**Comment by atzurk** [comment]
Hi there, I've used the sdcc version 3.8.0 with success in the setup.sh, in a M1 Max system. you can refer to issue #145

**Comment by baxpick** [comment]
Hello @lronaldo, maybe your macOS users with silicon can benefit from this: https://github.com/baxpick/cpctelera_example

@profludic @atzurk

**Comment by lronaldo** [comment]
Yeah, they probably will, as soon as I can take the time to reorganize branches, make 1.5 public and prepare new developments including SDCC 4.5

Thanks :)

**Comment by baxpick** [comment]
NP!

I plan also to upload `amd64` architecture  so other users are covered also.

And there is a secret plan to add support for Enterprise computers  :) (using code ported by Geco) but for this I will ask his permission and also will need some time to integrate.

Anyway, having cpctelera work without any need to install and compile it locally is very nice and image is quite small so all is very fast!

Looking forward to 1.5 and beyond!

**Comment by baxpick** [comment]
Now, on any architecture, you can easily build cpctelera projects:

1. Prepare

```bash
git clone -b development https://github.com/lronaldo/cpctelera.git

curl -L -o build.sh https://raw.githubusercontent.com/baxpick/cpctelera_example/main/build_cpctelera_project_using_container.sh
chmod +x build.sh

mkdir -p OUTPUT/CPC
mkdir -p OUTPUT/ENTERPRISE
```

2. Build

```bash
./build.sh \
    --folder-src ./cpctelera/examples/games/platformClimber \
    --folder-output OUTPUT/CPC \
    --platform cpc
```

<img width="874" alt="Image" src="https://github.com/user-attachments/assets/e2e56a76-99e5-43b9-b515-84d04e502dda" />

or even for Enterprise computer:

```bash
./build.sh \
    --folder-src ./cpctelera/examples/games/platformClimber \
    --folder-output OUTPUT/ENTERPRISE \
    --platform enterprise
```

![Image](https://github.com/user-attachments/assets/5fb905df-6702-48ad-9a77-5af63324d18d)

**Comment by pablo4licante** [comment]
Hi! I wanted to chime in since I tried installing and using CPCTelera on my Mac Mini M4 today, and I had some small details I wanted to share on top of what was mentioned in this thread.

First of all, it seems like the `wine` package on brew will be deprecated, and I found this repo explaining how to install it: https://github.com/Gcenx/wine-on-mac.

After that, I had to give the paths to my libraries and compiler binaries to the setup:

```bash
export CPATH="/opt/homebrew/include"
export LIBRARY_PATH="/opt/homebrew/lib"
export CFLAGS="-I/opt/homebrew/include"
export CXXFLAGS="-I/opt/homebrew/include"
export LDFLAGS="-L/opt/homebrew/lib"
```
And then I stumbled upon the problem mainly mentioned in this thread about the SDCC compatibility with the ARM version of Darwin. Firstly, I changed the sources of the compiler to the current version on GitHub: https://github.com/swegener/sdcc. I had no problems compiling, but when I tried to run any of the demos, they were completely broken. So, after revisiting the thread, I tried version 3.8.0 of the compiler (as mentioned by @atzurk), which I found here: https://sourceforge.net/projects/sdcc/files/sdcc/3.8.0. 
And after having to modify some of the source code (just converting some `(void *(*) (long)) xmalloc` to `(void *(*) (size_t)) xmalloc`), I got it installed :)

I also wanted to try the fix mentioned by @profludic, but the link seems to be inaccessible.

The CPCTelera setup assumed I was using a `.bash_profile` instead of `.zshrc`, but copying it to my config file was an easy fix.

With 3.8, I was able to compile all the demos and play the `platformClimber` demo for a bit with both RVM and Winape. Maybe it is a Wine issue, as I got great emulation in RVM, but on Winape I got pretty bad ghosting and tearing from the character movement.

So, for the moment, I will keep my development environment for Amstrad CPC in Linux, but I would love to see "first party" support for M-chip macOS in the next version of CPCTelera. Nonetheless, thanks @lronaldo for the great tooling and tutorials!

**Comment by baxpick** [comment]
@pablo4licante Hello, just to mention some useful info:

You can use docker images mentioned here in conversation ( and also here: https://github.com/baxpick/cpctelera_example ) - use precompiled cpctelera!

choose image you like from here:

https://hub.docker.com/r/braxpix/cpctelera-build-cpc/tags

tag development => cpctelera built from development branch
tag master => cpctelera built from master branch
tag next => cpctelera built from new port of cpctelera from https://github.com/Arnaud6128/cpctelera/tree/cpctelera_next

If you really want to build cpctelera on host system and not use prebuilt container, try this:

https://github.com/Arnaud6128/cpctelera/tree/cpctelera_next

it's already working (and in development) port of cpctelera to new SDCC and a few more stuff, check this: 

https://www.cpcwiki.eu/forum/programming/migration-cpctelera-to-sdcc-4-5-0/
